src/main/java/uk/ac/ebi/ampt2d/accession/commons/core/BasicAccessioningServiceSaveDelegate.java

```diff
@@ -0,0 +1,75 @@
+package uk.ac.ebi.ampt2d.accession.commons.core;
+
+import org.springframework.dao.DataIntegrityViolationException;
+import uk.ac.ebi.ampt2d.accession.commons.persistence.DatabaseService;
+import uk.ac.ebi.ampt2d.accession.commons.generators.ModelHashAccession;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Stack;
+
+/**
+ * This class is delegate for {@link BasicAccessioningService} to manage the save operation.
+ *
+ * The save operation can fail when an object has been already accessioned with a different value while having the
+ * same hash message and elements. In this case a database constraint exception can be raised. Due to limited
+ * information in those cases of error on JDBC driver, the save operation is implemented as an iterative version of a
+ * binary partition.
+ *
+ * If the save operation works, all elements have been stored, otherwise, because the save operation is expected to
+ * be transactional, the saved elements are dropped from database. And we split in two the batch of elements to save
+ * and try to save each part separately until all the split batches have been saved correctly and we find the specific
+ * elements that provoke error.
+ *
+ * @param <MODEL>
+ * @param <HASH>
+ * @param <ACCESSION>
+ */
+class BasicAccessioningServiceSaveDelegate<MODEL, HASH, ACCESSION> {
+
+    private class Partition {
+
+        private int start;
+        private int end;
+
+        public Partition(int start, int end) {
+            this.start = start;
+            this.end = end;
+        }
+    }
+
+    private final DatabaseService<MODEL, HASH, ACCESSION> dbService;
+
+    public BasicAccessioningServiceSaveDelegate(DatabaseService<MODEL, HASH, ACCESSION> dbService) {
+        this.dbService = dbService;
+    }
+
+    public SaveResponse<ACCESSION, MODEL> doSaveAccessions(
+            List<ModelHashAccession<MODEL, HASH, ACCESSION>> modelHashAccessions) {
+        Stack<Partition> partitions = new Stack<>();
+        partitions.add(new Partition(0, modelHashAccessions.size()));
```

Comment:
I think the concept of Partition(start,end) is closer to '[]' that to '[)', so I would find the code less confusing with close ends. I know is been added that way because sublist works that way, but I would find easier work with closed partitions an just do subList(partition.start, partition.end + 1). Anyway, if nobody else is bothered by this you can leave the code as it is now